src/batch/src/executor/delete.rs

```diff
@@ -84,7 +84,9 @@ impl DeleteExecutor {
 
             let chunk = StreamChunk::from_parts(vec![Op::Delete; len], data_chunk);
 
-            let notifier = source.write_chunk(chunk)?;
+            let notifier = source
+                .write_chunk(chunk)
+                .map_err(|err| anyhow::anyhow!(err))?;
```

Comment:
We may add a variant of `SourceError(BoxedError)` in the `ErrorCode`, and implement `From<SourceError>` for `RwError` so we can directly `?` here.

https://github.com/singularity-data/risingwave/blob/e89df1ea686335a2fd2b20281cc36f016e508af7/src/stream/src/executor/error.rs#L160-L165

After batch error is introduced, this will be `From<SourceError>` for `BatchError`.